frontend_tests/node_tests/stream_data.js

```diff
@@ -635,3 +635,27 @@ zrequire('marked', 'third/marked/lib/marked');
 
     assert.equal(page_params.notifications_stream, "foo");
 }());
+
+(function test_get_newbie_stream() {
+    var newbie = {
+        name: 'newbie',
+        stream_id: 234,
+        subscribed: true,
+        in_home_view: true,
+    };
+
+    var new_members = {
+        subscribed: true,
+        name: 'new members',
+        stream_id: 531,
+    };
+
+    assert.equal(stream_data.get_newbie_stream(), undefined);
+
+    stream_data.add_sub('newbie', newbie);
+    page_params.notifications_stream = 'newbie';
+    assert.equal(stream_data.get_newbie_stream(), 'newbie');
+
+    stream_data.add_sub('new members', new_members);
```

Comment:
Can we do a case where "new members" stream is not subscribed to?

Reply:
Added both cases.

Reply:
Submitted on GCI App. Commented PR but forgot to submit it 😄 .